core/field_multilineinput.js

```diff
@@ -91,6 +91,21 @@ Blockly.FieldMultilineInput.fromJson = function(options) {
   return new Blockly.FieldMultilineInput(text, undefined, options);
 };
 
+/**
+ * Serializes this field's value to XML. Should only be called by Blockly.Xml.
+ * @param {!Element} fieldElement The element to populate with info about the
+ *    field's state.
+ * @return {!Element} The element containing info about the field's state.
+ * @package
+ */
+Blockly.FieldMultilineInput.prototype.toXml = function(fieldElement) {
+  // Replace '\n' characters with html-escaped equivalent '&#10'. This is needed
+  // so the plain-text representation of the xml produced by `Blockly.Xml.domToText`
+  // will appear on a single line (this is a limitation of the plain-text format).
+  fieldElement.textContent = this.getValue().replace(/\n/g, '&#10');
```

Comment:
Oh this isn't quite right. Needs to be `&#10;`

Reply:
Shouldn't this use the original regex?

Reply:
I don't think it needs to, because the original regex was looking over all of the XML, whereas this one only gets the multi-line field's value.

Reply:
The original regexp was matching xml open/close tags. At this scope we're just operating on the content string (what appears between the xml tags).

Reply:
All right, and as a suggestion, what about wrapping the content in `<![CDATA[ ... ]]>` instead? Might be faster too.

Reply:
This change brings the performance of domToText to an acceptable level for MakeCode. We could try the CDATA approach if additional perf issues arise in serialization. I admit my understanding of when to use CDATA is spotty. I think it is useful when you want to embed XML in a content string.